payu/models/fms.py

```diff
@@ -76,9 +76,16 @@ def fms_collate(model):
     mpi = collate_config.get('mpi', False)
 
     if mpi:
-        # Must use envmod to be able to load mpi modules for collation
-        envmod.setup()
-        model.expt.load_modules()
+        # TODO: I've added module setup and load functions to
+        # Experiment.collate - as user modules needed to be use & loaded
+        # to expand executable paths. However, is loading mpi module
+        # for non-mpi collate jobs going to cause errors?
+        # If so, need to only use/load user modules in Experiment.collate
+        # and only load additional modules for mpi here
```

Comment:
I don't think it is a problem to load if not required, but best if we don't have to.

Reply:
I've split off loading the user modules from loading the other modules - so user modules will always get loaded as part of collate and `load_modules` - which loads mpi is only loaded if needed.